Added Delay to ActivityTypes to prevent need to use magic string

garypretty opened the pull request:
We have an context extension method which was using an activity type of "delay".  Given that this is part of the core SDK, I have brought this in line with other activity types and added it to the ActivityTypes class to prevent the need to use a magic string for comparison and hopefully avoid typo mistakes.

cleemullins commented:
I forgot to mention, that the underlying issue here is that **Delay** isn't yet added into the protocol itself. The client-side hacks you're looking at are symptoms of that. 

We plan to add it in, but it's in the backlog behind more important work...

garypretty commented:
Hi Chris. This makes sense, thanks for the context. How about we go for the ActivityTypesEx class for now and then we can just remove from that class as necessary when the swagger file gets updated. That way we get rid of the magic strings at least until the update and if the update doesn't happen soon then we have a way of handling more types if they are added in the meantime. If that makes sense then I'll update the PR later.

cleemullins commented:
@garypretty 

Thanks! That'll work perfectly.

garypretty commented:
Hi @cleemullins.  I have updated the PR and moved the "delay" type into an ActivityTypesEx as discussed, but I am referencing this directly at the moment, i.e. ActivityTypesEx.Delay, because I can't extent it as a partial due to AcitivityTypes not being a partial at present.  Is this something we can change easily? We can commit this now, but if we can make ActivityTypes a partial it would bring it into line with many other of the schema classes and allow us to extend it moving forward if we need to. 

Anyway, let me know if we are good to merge as we are or if you want to move anything around.

cleemullins commented:
Thanks Gary! This is perfect. 

I'll see what we can do about getting the Swagger file updated to generate the ActivityTypes as a partial class. In the short term, what's here is very reasonable.
